lib/spring/client/binstub.rb

```diff
@@ -38,7 +38,7 @@ class Binstub < Command
   require 'bundler'
 
   if (match = Bundler.default_lockfile.read.match(/^GEM$.*?^    (?:  )*spring \((.*?)\)$.*?^$/m))
-    Gem.paths = { 'GEM_PATH' => [Bundler.bundle_path.to_s, *Gem.path].uniq }
+    Gem.paths = { 'GEM_PATH' => [Bundler.bundle_path.to_s, *Gem.path].uniq.join(Gem.path_separator) }
```

Comment:
Apparently `Gem.path_separator` might not be a string: https://github.com/rubygems/rubygems/pull/1476